src/test/java/de/tum/cit/aet/artemis/programming/ProgrammingExerciseIntegrationTestService.java

```diff
@@ -2319,7 +2319,14 @@ void test_redirectGetTemplateRepositoryFilesWithContent() throws Exception {
                 fail("Setup template threw unexpected exception: " + e.getMessage());
             }
             return localRepository;
-        });
+        };
+
+        if (omitBinaries) {
+            test_redirectGetTemplateRepositoryFilesWithContentOmitBinary(redirectFnc);
+        }
+        else {
+            test_redirectGetTemplateRepositoryFilesWithContent(redirectFnc);
+        }
```

Comment:
I think it would be nicer to simply split this into two functions, and directly call them. (instead of calling the same function with different arguments